gp-includes/things/original.php

```diff
@@ -107,8 +107,15 @@ public function import_for_project( $project, $translations ) {
 
 		$possibly_added = $possibly_dropped = array();
 
-		foreach( $translations->entries as $entry ) {
+		foreach( $translations->entries as $key => $entry ) {
 			$wpdb->queries = array();
+
+			// Context needs to match VARCHAR(255) in the database schema.
+			if ( gp_strlen( $entry->context ) > 255 ) {
+				$entry->context = gp_substr( $entry->context, 0, 255 );
+				$translations->entries[ $entry->key() ] = $entry;
```

Comment:
We do need to assign the entire `$entry` here, because trimming `$entry->context` changes `$entry->key()`, and the `array_key_exists( $key, $translations->entries )` check fails later.

Without this assignment, the affected string ends up in `$possibly_dropped` and gets repeatedly removed / re-added on each subsequent import.